metagpt/utils/repair_llm_raw_output.py

```diff
@@ -198,6 +206,21 @@ def repair_invalid_json(output: str, error: str) -> str:
             new_line = line.replace("}", "")
         elif line.endswith("},") and output.endswith("},"):
             new_line = line[:-1]
+        # remove comments in output json str, after json value content, maybe start with #, maybe start with //
+        elif rline[col_no] == "#" or rline[col_no] == "/":
+            new_line = rline[:col_no]
+            for i in range(line_no + 1, len(arr)):
+                # look for # or // comments and make sure they are not inside the string value
+                comment_index = -1
+                for match in re.finditer(r"(\".*?\"|\'.*?\')|(#|//)", line):
+                    if match.group(1):  # if the string value
+                        continue
+                    if match.group(2):  # if comments
+                        comment_index = match.start(2)
+                        break
+                # if comments, then delete them
+                if comment_index != -1:
+                    arr[i] = arr[i][:comment_index].rstrip()
```

Comment:
The new two pieces of logic are the same, would it be better to merge them?

Reply:
Okay, it's a good idea, I merged the code and resubmitted it. Please have a look, thanks.